Source/ACE.Entity/AceObject.cs

```diff
@@ -1816,24 +1816,48 @@ public object Clone(uint guid)
         {
             AceObject ret = (AceObject)Clone();
             ret.AceObjectId = guid;
+            // We are cloning a new AceObject with a new AceObjectID - need to set this to false. Og II
+            ret.HasEverBeenSavedToDatabase = false;
+            ret.IsDirty = true;
+
+            ret.PaletteOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.TextureOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.AnimationOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.IntProperties.ForEach(c => c.AceObjectId = guid);
+            ret.Int64Properties.ForEach(c => c.AceObjectId = guid);
+            ret.DoubleProperties.ForEach(c => c.AceObjectId = guid);
+            ret.BoolProperties.ForEach(c => c.AceObjectId = guid);
+            ret.DataIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.InstanceIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.StringProperties.ForEach(c => c.AceObjectId = guid);    
+            ret.GeneratorLinks.ForEach(c => c.AceObjectId = guid);
+            
+            // We're not seemingly changing the guids for the Dictionaries below from Clone(), is this bad?
```

Comment:
In general I dislike the fact that save-time data creeps into our live data-structures, and the smell here is associated with the presence of the redundant data.  I believe either implementing the guid update at clone time (as Ripley has done here), or at save time is sufficient.  Other than the fact I don't like this data existing at all, I think Ripley's implementation here is fine.

It looks like you and Ripley have essentially created the same code to accomplish this, and it seems to be a fine way to go about doing it to me.